lib/spack/spack/hooks/__init__.py

```diff
@@ -44,11 +41,18 @@ def __init__(self, hook_name):
     def _populate_hooks(cls):
         # Lazily populate the list of hooks
         cls._hooks = []
-        relative_names = list(llnl.util.lang.list_modules(spack.paths.hooks_path))
 
-        # We want this hook to be the last registered
-        relative_names.sort(key=lambda x: x == "write_install_manifest")
-        assert relative_names[-1] == "write_install_manifest"
+        # Since the order is important, we just fix it here, it's not like
+        # hooks get added every other day.
```

Comment:
I'm fine with having this be a static list, but perhaps have a test or audit that checks it's up to date?

Reply:
Is this comment still relevant?  I think it's not since you now use `ensure_last`